src/type.js

```diff
@@ -4,64 +4,38 @@ import { core } from 'metal';
 
 const validators = {
 	any: () => true,
-	array: validateType('array'),
-	bool: validateType('boolean'),
-	func: validateType('function'),
-	number: validateType('number'),
-	object: validateType('object'),
-	string: validateType('string'),
-
-	arrayOf: (validator) => {
-		return composeValidator(
-			arrayOfValidatorFactory,
-			validator
-		);
-	},
-	instanceOf: (expectedClass) => {
-		return composeValidator(
-			instanceOfValidatorFactory,
-			expectedClass
-		);
-	},
-	objectOf: (validator) => {
-		return composeValidator(
-			objectOfValidatorFactory,
-			validator
-		);
-	},
-	oneOfType: (arrayOfTypeValidators) => {
-		return composeValidator(
-			oneOfTypeValidatorFactory,
-			arrayOfTypeValidators
-		);
-	},
-	shape: (shape) => {
-		return composeValidator(
-			shapeValidatorFactory,
-			shape
-		);
-	}
+	array: primitiveTypeValidatorFactory('array'),
```

Comment:
For the primitive type validators, I think the best way to do that is by just calling `array: validateType('array');` otherwise, when a user is using a validator that have actually invoke the method rather than just returning a function.

For example:
```js
const validators = {
  array: function() {
    return validateType('array');
  }
}

const validators2 = {
  array: validateType('array')
}

Component.STATE = {
     key1: {
           validator: validators.array()
     },
     key2: {
           validator: validators2.array
     }
}
```

Unless you are thinking that validateType would now take 4 arguments, `validateType(type, value, name, context)`

As far as inlining the other factories, that is fine by me.